include/tvm/ffi/function.h

```diff
@@ -166,21 +171,19 @@ class FunctionObjImpl : public FunctionObj {
 
   /*!
    * \brief Derived object class for constructing ffi::FunctionObj.
-   * \param callable The type-erased callable object (rvalue).
-   */
-  explicit FunctionObjImpl(TCallable&& callable) : callable_(std::move(callable)) {
-    this->safe_call = SafeCall;
-    this->cpp_call = reinterpret_cast<void*>(CppCall);
-  }
-  /*!
-   * \brief Derived object class for constructing ffi::FunctionObj.
-   * \param callable The type-erased callable object (lvalue).
+   * \param args The arguments to construct TCallable
    */
-  explicit FunctionObjImpl(const TCallable& callable) : callable_(callable) {
+  template <typename... Args>
+  explicit FunctionObjImpl(Args&&... args) : callable_(std::forward<Args>(args)...) {
     this->safe_call = SafeCall;
     this->cpp_call = reinterpret_cast<void*>(CppCall);
   }
 
+  FunctionObjImpl(const FunctionObjImpl&) = delete;
+  FunctionObjImpl& operator=(const FunctionObjImpl&) = delete;
+
+  TCallable& GetCallable() { return callable_; }
```

Comment:
nit: For some reason our convention is to always return pointers instead of references, so it would be better if we do

```suggestion
  TCallable* GetCallable() { return &callable_; }
```